leader-election-impl/src/main/java/com/palantir/paxos/SqlitePaxosStateLogMigrationState.java

```diff
@@ -54,29 +56,45 @@ private void initialize() {
     }
 
     public void migrateToValidationState() {
-        execute(dao -> dao.migrateToVersion(namespace, useCase, States.VALIDATION.schemaVersion));
+        execute(migrateToState(States.VALIDATION));
     }
 
     public void migrateToMigratedState() {
-        execute(dao -> dao.migrateToVersion(namespace, useCase, States.MIGRATED.schemaVersion));
+        execute(migrateToState(States.MIGRATED));
     }
 
     public boolean hasMigratedFromInitialState() {
-        return !Objects.equals(States.NONE.getSchemaVersion(), execute(dao -> dao.getVersion(namespace, useCase)));
+        return execute(dao -> dao.getVersion(namespace, useCase).isPresent());
     }
 
     public boolean isInValidationState() {
-        return States.VALIDATION.getSchemaVersion().equals(execute(dao -> dao.getVersion(namespace, useCase)));
+        return execute(dao -> dao.getVersion(namespace, useCase)
+                .map(States.VALIDATION.getSchemaVersion()::equals)
+                .orElse(false));
     }
 
     public boolean isInMigratedState() {
-        return States.MIGRATED.getSchemaVersion().equals(execute(dao -> dao.getVersion(namespace, useCase)));
+        return execute(dao -> dao.getVersion(namespace, useCase)
+                .map(States.MIGRATED.getSchemaVersion()::equals)
+                .orElse(false));
     }
 
     private <T> T execute(Function<Queries, T> call) {
         return jdbi.withExtension(Queries.class, call::apply);
     }
 
+    private Function<Queries, Boolean> migrateToState(States state) {
+        return dao -> {
+            assertCurrentStateAtMost(dao, state);
+            return dao.migrateToVersion(namespace, useCase, state.getSchemaVersion());
+        };
+    }
+
+    private void assertCurrentStateAtMost(Queries dao, States state) {
+        dao.getVersion(namespace, useCase).ifPresent(currentVersion ->
+                Preconditions.checkState(currentVersion <= state.getSchemaVersion()));
```

Comment:
Could we add a bit more information (e.g. namespace, use-case, current/intended version) as args?